hassio/src/update-available/update-available-card.ts

```diff
@@ -0,0 +1,381 @@
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false }) public supervisor!: Supervisor;
+
+  @property({ attribute: false }) public route!: Route;
+
+  @property({ type: Boolean }) public narrow!: boolean;
+
+  @property({ attribute: false }) public addonSlug?: string;
+
+  @state() private _updateEntry?: string;
+
+  @state() private _changelogContent?: string;
+
+  @state() private _addonInfo?: HassioAddonDetails;
+
+  @state() private _createBackup = true;
+
+  @state() private _action: "backup" | "update" | null = null;
+
+  @state() private _error?: string;
+
+  private _isAddon = false;
+
+  private _addonStoreInfo = memoizeOne(
+    (slug: string, storeAddons: StoreAddon[]) =>
+      storeAddons.find((addon) => addon.slug === slug)
+  );
+
+  protected render(): TemplateResult {
+    if (!this._updateEntry) {
+      return html``;
+    }
+    const name =
+      // @ts-ignore
+      this._addonInfo?.name || SUPERVISOR_UPDATE_NAMES[this._updateEntry];
+    const changelog = !this._isAddon
+      ? changelogUrl(
+          this.hass,
+          this._updateEntry,
+          this.supervisor[this._updateEntry]?.version
+        )
+      : undefined;
+    return html`
+      <ha-card
+        .header=${this.supervisor.localize("update_available.update_name", {
+          name,
+        })}
+      >
+        <div class="card-content">
+          ${this._error
+            ? html`<ha-alert alert-type="error">${this._error}</ha-alert>`
+            : ""}
+          ${this._action === null
+            ? html`
+                ${this._changelogContent
+                  ? html`
+                      <ha-expansion-panel header="Changelog" outlined>
+                        <ha-markdown .content=${this._changelogContent}>
+                        </ha-markdown>
+                      </ha-expansion-panel>
+                    `
+                  : ""}
+                <div class="versions">
+                  <p>
+                    ${this.supervisor.localize("update_available.description", {
+                      name,
+                      version:
+                        this._addonInfo?.version ||
+                        this.supervisor[this._updateEntry]?.version,
+                      newest_version:
+                        this._addonInfo?.version_latest ||
+                        this.supervisor[this._updateEntry]?.version_latest,
+                    })}
+                  </p>
+                </div>
+                ${!["os", "supervisor"].includes(this._updateEntry)
+                  ? html`
+                      <ha-formfield
+                        .label=${this.supervisor.localize(
+                          "update_available.create_backup"
+                        )}
+                      >
+                        <ha-checkbox
+                          .checked=${this._createBackup}
+                          @click=${this._toggleBackup}
```

Comment:
Since clicking has no immediate effect, no need to store it as an instance variable. Instead read the value from the `<ha-checkbox>` when you need to know if it's checked

```ts
    get _shouldCreateBackup() {
      return this.shadowRoot!.querySelector("ha-checkbox")!.checked;
    }
```